src/main/java/codechicken/nei/recipe/GuiCraftingRecipe.java

```diff
@@ -24,12 +24,13 @@ public static boolean openRecipeGui(String outputId, Boolean overlay, Object...
         Minecraft mc = NEIClientUtils.mc();
 
         BookmarkRecipeId recipeId =
```

Comment:
Hey so this feature works and I think it's a big improvement, but I'm not sure I'm doing the right in `openRecipeGui` when it comes to interacting with NEIClientConfig.saveCurrentRecipeInBookmarksEnabled() and with the overlay feature (no idea what that is or how to make it work).

Reply:
Also it looks weird to me what the overlay handling code has you call displayGuiScreen with a new gui, but then call it again with the previous gui. Wouldn't it just be better to not change screens at all?

Reply:
about overlay ask @MuXiu1997

Reply:
Okay someone pointed me to overlay's functionality, I'll see if I need to clean up anything.